src/common/snippets/include/snippets/pass/split_dimension_m.hpp

```diff
@@ -67,7 +67,18 @@ class SplitDimensionM: public CommonOptimizations::SubgraphPass {
 
 private:
     static std::shared_ptr<ov::op::v0::MatMul> get_matmul(const std::shared_ptr<op::Subgraph>& subgraph);
-    static std::pair<size_t, size_t> get_splited_dimensions(size_t batch_dim, size_t m_dim, size_t optimal_parallelism_work_amount);
+    /**
+     * @brief Contains splitM approaches allowing to get the batch ideally divisible by optimal_parallelism_work_amount
+     */
+    static std::pair<size_t, size_t> compute_ideal_cases_heuristic(size_t batch_dim, size_t m_dim, size_t optimal_parallelism_work_amount);
+    /**
+     * @brief Aggressively splits m_dim to minimize kernel_m in order to reduce waiting time for idle threads at the last parallel loop iteration.
+     */
+    static std::pair<size_t, size_t> compute_aggressive_heuristic(size_t batch_dim, size_t m_dim, size_t optimal_parallelism_work_amount);
+    /**
+     * @brief Conservatively splits m_dim to get the batch in (optimal_parallelism_work_amount, 2 * optimal_parallelism_work_amount) interval
+     */
+    static std::pair<size_t, size_t> compute_conservative_heuristic(size_t batch_dim, size_t m_dim, size_t optimal_parallelism_work_amount);
```

Comment:
A few minor comments with respect to the methods' names:
1) the functions try to split M dimension into two: batch_m and new_m, so it seems logical to call them smth like `get_splitted_*` or `split_m_*` or `compute_m_split` etc. Heuristic is used inside this functions to perform the split, so the functions do not compute heuristic, but rather use it to compute smth else (split).
2) We should try to use more descriptive names instead of `aggressive` or `conservative`, because what seems conservative to one may seem aggressive to the other. Possible options include: `split_ideally`, `split_max_kernel_work` and `split_max_parallel_work`

Reply:
I tried to apply your suggestions. The only left thing is `split_conservatively_increase_parallel_wa`: "conservatively" is still ambiguous, but I have no idea what adverb reflects the function's description better, so if you have any ideas please let me know

Reply:
Probably `split_fallback_increase_parallel_wa`, since it's used as a last resort to increase parallel wa? Or `split_default_...` also an option here